mungegithub/mungers/milestone-maintainer.go

```diff
@@ -0,0 +1,461 @@
+		priorityCriticalUrgent:        "Never automatically move out of a release milestone; continually escalate to contributor and SIG through all available channels.",
+		"priority/important-soon":     "Escalate to the issue owners and SIG owner; move out of milestone after several unsuccessful escalation attempts.",
+		"priority/important-longterm": "Escalate to the issue owners; move out of the milestone after 1 attempt.",
+	}
+
+	milestoneRemovedLabel          = strings.ToLower(strings.Replace(milestoneRemoved, " ", "-", -1))
+	milestoneLabelsIncompleteLabel = strings.ToLower(strings.Replace(milestoneLabelsIncomplete, " ", "-", -1))
+	milestoneLabelsCompleteLabel   = strings.ToLower(strings.Replace(milestoneLabelsComplete, " ", "-", -1))
+	milestoneLabels                = []string{milestoneRemovedLabel, milestoneLabelsCompleteLabel, milestoneLabelsIncompleteLabel}
+)
+
+// milestoneNotification configures the active notification for the munger
+type milestoneNotification struct {
+	deleteExisting bool
+	description    string
+	message        string
+}
+
+// MilestoneMaintainer enforces the process for sheperding issues into the release.
+type MilestoneMaintainer struct {
+	botName  string
+	features *features.Features
+
+	activeMilestone string
+}
+
+func init() {
+	RegisterMungerOrDie(&MilestoneMaintainer{})
+}
+
+// Name is the name usable in --pr-mungers
+func (m *MilestoneMaintainer) Name() string { return "milestone-maintainer" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (m *MilestoneMaintainer) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (m *MilestoneMaintainer) Initialize(config *github.Config, features *features.Features) error {
+	if len(m.activeMilestone) == 0 {
+		return errors.New("active-milestone must be supplied")
+	}
+	m.botName = config.BotName
+	m.features = features
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (m *MilestoneMaintainer) EachLoop() error { return nil }
+
+// RegisterOptions registers options for this munger; returns any that require a restart when changed.
+func (m *MilestoneMaintainer) RegisterOptions(opts *options.Options) sets.String {
+	opts.RegisterString(&m.activeMilestone, "active-milestone", "", "The active milestone that this munger will maintain issues for.")
+	return sets.NewString("active-milestone")
+}
+
+// Munge is the workhorse the will actually make updates to the issue
+func (m *MilestoneMaintainer) Munge(obj *github.MungeObject) {
+	if m.ignoreObject(obj) {
+		return
+	}
+
+	comment, ok := latestNotificationComment(obj, m.botName)
+	if !ok {
+		return
+	}
+
+	labelName, notifyState := notificationState(obj, comment, m.botName)
+	// Always attempt to ensure the milestone label
+	if len(labelName) > 0 && !ensureLabel(obj, labelName) {
+		return
+	}
+	if notifyState == nil {
+		// No notification change is required
+		return
+	}
+
+	if notifyState.deleteExisting {
```

Comment:
I think the `deleteExisting` field is unnecessary. Just delete the comment if the comment is non-nil and `notifyState` is non-nil (we know the later to be true by the time we reach this line).